### PR TITLE
taskwarrior: Use inotify to get instant changes

### PR DESCRIPTION
**Pull request by cfsmp3**
Pretty much copy/pasted the Thread block from backlight, not a lot to it. Watch for changes on ~/.task and update the taskwarrior block if needed.

I haven't removed the polling code.

To test, just create a task:
```
task add "Fake task"
```
And you'll see the block updating instantly.

### CODE REVIEW

**Comment by ammgws** [comment]
We don't actually need to use the dirs crate, do we? IIRC we already have a helper function, or it was done trivially per block. Can you have a look?

Also why is interval still around?

On mobile so only had a brief look

**Comment by MaxVerevkin** [comment]
Should the path be configurable?

**Comment by cfsmp3** [comment]
> We don't actually need to use the dirs crate, do we? IIRC we already have a helper function, or it was done trivially per block. Can you have a look?

What we have in notmuch.rs is this:

```
#[allow(deprecated)]
let home_dir = match env::home_dir() {
```
And:

 Deprecated since 1.29.0:

This function’s behavior is unexpected and probably not what you want. Consider using a crate from crates.io instead.

dirs seems to have the sanest implementation (I did look).

> Also why is interval still around?

From inotify's documentation:

https://man7.org/linux/man-pages/man7/inotify.7.html

```
 Limitations and caveats
[...]
 Inotify reports only events that a user-space program triggers
       through the filesystem API.  As a result, it does not catch
       remote events that occur on network filesystems.  
[...]
       The inotify API does not report file accesses and modifications
       that may occur because of mmap(2), msync(2), and munmap(2).
[...]
 Note that the event queue can overflow.  In this case, events are
       lost.  Robust applications should handle the possibility of lost
       events gracefully.  For example, it may be necessary to rebuild
       part or all of the application cache. 
```
I believe when you put all of that together it's sane to keep some polling (maybe we can increase the default interval though).

**Comment by cfsmp3** [comment]
> Should the path be configurable?

Yes - because it is in taskwarrior itself.
Done.

**Comment by MaxVerevkin** [comment]
> From inotify's documentation:

I'm surprised it works for backlight block, even though It uses inotify to watch sysfs, which, AFAIK, is not supported at all and is considered to be a misuse.

File: src/blocks/taskwarrior.rs [hunk 1]
```diff
@@ -80,10 +83,20 @@ pub struct TaskwarriorConfig {
...
+    pub data_location: String,
 }
 
 impl Default for TaskwarriorConfig {
     fn default() -> Self {
+        let home_dir = match dirs::home_dir() {
+            Some(path) => path.into_os_string().into_string().unwrap(),
+            None => "".to_owned(),
+        };
+        let task_dir = format!("{}/.task", home_dir);
```
Comment (MaxVerevkin) [inline src/blocks/taskwarrior.rs:98]: `shellexpand` will expand `~` as a home directory anyway, so I suppose you can just do `let task_dir = "~/.task"`.

Comment (cfsmp3) [reply src/blocks/taskwarrior.rs]: > `shellexpand` will expand `~` as a home directory anyway, so I suppose you can just do `let task_dir = "~/.task"`.

That's better. Done.

**Comment by MaxVerevkin** [comment]
The interval was already high, so I think everything is good. Thanks!

**Pull request merged**